go-controller/pkg/ovn/egressgw.go

```diff
@@ -189,6 +189,22 @@ func (oc *Controller) addPodExternalGWForNamespace(namespace string, pod *kapi.P
 	if err != nil {
 		return fmt.Errorf("failed to ensure namespace locked: %v", err)
 	}
+	// build temporary ipTracker map to look for duplicate IPs
+	// key: IP, value: pod GW name
+	ipTracker := make(map[string]string)
+	for podName, gwInfo := range nsInfo.routingExternalPodGWs {
+		for _, gwIP := range gwInfo.gws {
+			ipTracker[gwIP.String()] = podName
+		}
+	}
+
+	for _, gwIP := range egress.gws {
+		if podName, ok := ipTracker[gwIP.String()]; ok {
+			nsUnlock()
+			return fmt.Errorf("unable to add pod: %s/%s as external gateway. This pod has a duplicate gateway "+
+				"IP address as pod: %q. IP address: %q", namespace, pod.Name, podName, gwIP.String())
```

Comment:
this should be pod.namespace, pod.Name and I should also indicate in the log which namespace this pod was trying to serve.